runtime-modules/content/src/tests/fixtures.rs

```diff
@@ -0,0 +1,932 @@
+use crate::*;
+use frame_support::assert_ok;
+use frame_support::traits::Currency;
+use sp_runtime::DispatchError;
+
+// constants used
+pub const VOUCHER_OBJECTS_NUMBER_LIMIT: u64 = 40;
+pub const VOUCHER_OBJECTS_SIZE_LIMIT: u64 = 400;
+pub const STORAGE_BUCKET_OBJECTS_NUMBER_LIMIT: u64 = 10;
+pub const STORAGE_BUCKET_OBJECTS_SIZE_LIMIT: u64 = 100;
+pub const STORAGE_BUCKET_ACCEPTING_BAGS: bool = true;
+pub const DATA_OBJECTS_NUMBER: u8 = 2;
+
+// type aliases
+type AccountId = <Test as frame_system::Trait>::AccountId;
+type VideoId = <Test as Trait>::VideoId;
+
+// fixtures
+pub struct CreateChannelFixture {
+    sender: AccountId,
+    actor: ContentActor<CuratorGroupId, CuratorId, MemberId>,
+    params: ChannelCreationParameters<Test>,
+}
+
+impl CreateChannelFixture {
+    pub fn default() -> Self {
+        Self {
+            sender: DEFAULT_MEMBER_ACCOUNT_ID,
+            actor: ContentActor::Member(DEFAULT_MEMBER_ID),
+            params: ChannelCreationParameters::<Test> {
+                assets: None,
+                meta: None,
+                reward_account: None,
+                collaborators: BTreeSet::new(),
+            },
+        }
+    }
+
+    pub fn with_sender(self, sender: AccountId) -> Self {
+        Self { sender, ..self }
+    }
+
+    pub fn with_actor(self, actor: ContentActor<CuratorGroupId, CuratorId, MemberId>) -> Self {
+        Self { actor, ..self }
+    }
+
+    pub fn with_assets(self, assets: StorageAssets<Test>) -> Self {
+        Self {
+            params: ChannelCreationParameters::<Test> {
+                assets: Some(assets),
+                ..self.params
+            },
+            ..self
+        }
+    }
+
+    pub fn with_collaborators(self, collaborators: BTreeSet<MemberId>) -> Self {
+        Self {
+            params: ChannelCreationParameters::<Test> {
+                collaborators: collaborators,
+                ..self.params
+            },
+            ..self
+        }
+    }
+
+    pub fn with_reward_account(self, reward_account: AccountId) -> Self {
+        Self {
+            params: ChannelCreationParameters::<Test> {
+                reward_account: Some(reward_account),
+                ..self.params
+            },
+            ..self
+        }
+    }
+
+    pub fn call_and_assert(&self, expected_result: DispatchResult) {
+        let origin = Origin::signed(self.sender.clone());
+        let balance_pre = Balances::usable_balance(self.sender);
+        let channel_id = Content::next_channel_id();
+        let channel_bag_id = Content::bag_id_for_channel(&channel_id);
+        let beg_obj_id = storage::NextDataObjectId::<Test>::get();
+        let actual_result =
+            Content::create_channel(origin, self.actor.clone(), self.params.clone());
+        let end_obj_id = storage::NextDataObjectId::<Test>::get();
+
+        assert_eq!(actual_result, expected_result);
+
+        let balance_post = Balances::usable_balance(self.sender);
+
+        match actual_result {
```

Comment:
Could be simplified: if actual_result.is_ok()

Reply:
I have added a hopefully cleaner solution

Reply:
fixed in ebd78029211e0b9911592cbf28c406e8f9727ff4